src/delfino/main.py

```diff
@@ -83,17 +89,74 @@ def invoke(self, ctx: click.Context) -> Any:
     def get_help(self, *args, **kwargs) -> str:
         help_str = super().get_help(*args, **kwargs)
 
-        if self._hidden_plugins:
+        if self._hidden_commands:
             if logging.root.level == logging.DEBUG:
                 help_str += click.style(
-                    f"\n\nDisabled command{'s' if len(self._hidden_plugins) > 1 else ''}: "
-                    + ", ".join(sorted(self._hidden_plugins))
+                    f"\n\nDisabled command{'s' if len(self._hidden_commands) > 1 else ''}: "
+                    + ", ".join(sorted(self._hidden_commands))
                     + f" (see 'tool.{ENTRY_POINT}.disable_commands' in '{PYPROJECT_TOML_FILENAME}')",
                     fg="white",
                 )
 
         return help_str
 
+    def _get_commands(self) -> Dict[str, click.Command]:
+        if self._commands:
+            return self._commands
+
+        # default
+        self._commands.update(find_commands(self.PRESET_COMMAND_PACKAGE, required=True))
+
+        # external
+        self._commands.update(self.ExternalCommandsLoader(self._hidden_plugin_commands).load())
+
+        # local
+        self._commands.update(find_commands(COMMANDS_DIRECTORY_NAME, required=False))
+        self._commands.update(find_commands("tasks", required=False, new_name=COMMANDS_DIRECTORY_NAME))
+
+        return self._commands
+
+    class ExternalCommandsLoader:
```

Comment:
You showed me in person that you based this functionality on an official guide. Please add a link into the docstring of this class.

I have to say that the functionality is rather obscure. When looking at `load()` + `_load_plugins()` + `_load_commands_for_plugin()`, they have a high cyclomatic complexity. It is basically 4 nested `for` loops.
Things to consider:
- Collapsing the three functions into one (could make it more readable).
- Type annotating `plugin`, `entry_point` and `package`
- I see iteration over `entry_point` twice. Any chance it could be only once?

Reply:
The are 4 nested loops and that is why I've separated them into methods to describe each part by method name as I couldn't come up with a solution to reduce the loops.


> Collapsing the three functions into one (could make it more readable).
> Type annotating plugin, entry_point and package
> I see iteration over entry_point twice. Any chance it could be only once?

I'll do this with some comments.

Reply:
I've refactored the code as https://github.com/radeklat/delfino/pull/32#discussion_r977049892.

@radeklat please take a look it again 🙏